src/PHPStan/GetAttributeCheck.php

```diff
@@ -75,35 +78,103 @@ public function check(StaticCall $node, Scope $scope): ?array
             $typeStrings = $scope->getType($arguments['type'])->getConstantScalarTypes();
 
             if (count($typeStrings) !== 1) {
-                return [new MixedType(), $errors];
+                return [new MixedType(), []];
             }
 
             $type = $typeStrings[0]->getValue();
         }
 
-        // vendor/twig/twig/src/Extension/CoreExtension.php:1643
-        if ($type !== Template::METHOD_CALL) {
-            if ($objectType->isConstantArray()->yes()) {
-                if ($objectType->hasOffsetValueType($propertyOrMethodType)->yes()) {
-                    return [$objectType->getOffsetValueType($propertyOrMethodType), $errors];
+        if ($objectType instanceof UnionType) {
+            $subTypeResults = [];
+            $subTypeErrors = [];
+            foreach ($objectType->getTypes() as $subTypes) {
+                $result = $this->checkSingleType(
+                    $type,
+                    $subTypes,
+                    $propertyOrMethodType,
+                    $propertyOrMethod,
+                    $scope,
+                    $arguments['args'],
+                );
+
+                if (!$result[0] instanceof ErrorType) {
+                    $subTypeResults[] = $result[0];
+                }
+                $subTypeErrors[] = $result[1];
+            }
+
+            $reportMaybe = true; // TODO: Inject the reportMaybe property.
+            if ($subTypeResults === []) {
+                $errors = [
+                    RuleErrorBuilder::message(sprintf(
+                        'Neither the property "%1$s" nor one of the methods "%1$s()", "get%1$s()", "is%1$s()", "has%1$s()" or "__call()" exist and have public access in class "%2$s".',
+                        $propertyOrMethod,
+                        $objectType->describe(VerbosityLevel::typeOnly()),
+                    ))
+                    ->identifier('getAttribute.notFound')
+                    ->build()
+                ];
+            } elseif ($reportMaybe) {
+                $errorBuilder = RuleErrorBuilder::message(sprintf(
+                    'TODO Might not exists',
```

Comment:
Any suggestion for the message ?

The idea is to report call which might fail like 
`format` on DateTime|null. And in the tip, I least the reason of the possible failure.